conda_build/build.py

```diff
@@ -475,27 +801,90 @@ def write_hash_input(m):
+    # It is nonsensical to replace anything in a symlink.
+    files = [f for f in files_in if not os.path.islink(os.path.join(prefix, f))]
+    all_matches = {}
+    ignore_files = m.ignore_prefix_files()
+    ignore_types = set()
+    if not hasattr(ignore_files, "__iter__"):
+        if ignore_files is True:
+            ignore_types.update((FileMode.text.name, FileMode.binary.name))
+        ignore_files = []
+    if (not m.get_value('build/detect_binary_files_with_prefix', True) and
+        not m.get_value('build/binary_has_prefix_files', None)):
+        ignore_types.update((FileMode.binary.name,))
+    files_with_prefix = [(prefix_placeholder, FileMode.binary.name if \
+                         open(os.path.join(prefix, f), 'rb+').read().find(b'\x00') != -1 else \
+                             FileMode.text.name, f) for f in files]
+    ignore_files.extend(
+        f[2] for f in files_with_prefix if (f[1] in ignore_types and \
+        f[2] not in ignore_files) or prefix_replacement_excluded(os.path.join(prefix, f[2])))
+    files_with_prefix = [f for f in files_with_prefix if f[2] not in ignore_files]
+
+    all_matches = have_regex_files([f[2] for f in files_with_prefix], prefix=prefix,
+                                   tag='prefix',
+                                   regex_re=prefix.encode('utf-8'),
+                                   replacement_re=prefix_placeholder.encode('utf-8'),
+                                   also_binaries=True,
+                                   match_records=all_matches)
+    files_with_prefix = [f for f in files_with_prefix if f[2] in all_matches]
+    ext = '.pc'
+    all_matches = have_regex_files(files=[f for f in files if f.endswith(ext)], prefix=prefix,
+                                   tag='pkg-config build metadata',
+                                   regex_re=b'(?:-L|-I)?\"?([^;\s]+\/sysroot\/)',
+                                   replacement_re=b'$(CONDA_BUILD_SYSROOT_S)',
+                                   match_records=all_matches,
+                                   regex_rg=b'([^;\s"]+/sysroot/)')
+    ext = '.cmake'
+    all_matches = have_regex_files(files=[f for f in files if f.endswith(ext)], prefix=prefix,
+                                   tag='CMake build metadata',
+                                   regex_re = b'([^;\s"]+/sysroot)',
+                                   replacement_re = b'$ENV{CONDA_BUILD_SYSROOT}',
+                                   match_records = all_matches)
+    ext = ('.pri', '.prl')
+    all_matches = have_regex_files(files=[f for f in files if f.endswith(ext)], prefix=prefix,
+                                   tag='qmake build metadata',
+                                   regex_re=b'(?:-L|-I)?\"?([^;\s]+\/sysroot)',
+                                   replacement_re=b'$(CONDA_BUILD_SYSROOT)',
+                                   match_records = all_matches,
+                                   regex_rg=b'([^;\s"]+/sysroot)')
```

Comment:
This could/should be configurable, probably in conda_build_config.yaml to allow arbitrary replacement.